pkg/client/client.go

```diff
@@ -65,11 +67,14 @@ func New(cfg Config) (*CortexClient, error) {
 		client = http.Client{Transport: transport}
 	}
 
+	legacy := cfg.UseLegacyRoutes == "true"
+
 	return &CortexClient{
 		key:      cfg.Key,
 		id:       cfg.ID,
 		endpoint: endpoint,
 		client:   client,
+		legacy:   legacy,
 	}, nil
```

Comment:
**suggestion:** similar to above, use a string to store the path instead of a boolean.

```suggestion
	path := rulerAPIPath
	if cfg.UseLegacyRoutes {
		path = legacyAPIPath
	}

	return &CortexClient{
		key:      cfg.Key,
		id:       cfg.ID,
		endpoint: endpoint,
		client:   client,
		apiPath:  path,
	}, nil
```